components/salsa-2022/src/input_field.rs

```diff
@@ -0,0 +1,67 @@
+use std::hash::Hash;
+use rustc_hash::FxHashMap;
+use crate::{Durability, IngredientIndex, Revision, Runtime};
+use crate::cycle::CycleRecoveryStrategy;
+use crate::ingredient::Ingredient;
+use crate::key::DependencyIndex;
+use crate::runtime::local_state::QueryInputs;
+use crate::runtime::StampedValue;
+
+pub struct InputFieldIngredient<K, F> {
```

Comment:
Let's add a comment here, e.g.

```suggestion
/// Ingredient used to represent the fields of a `#[salsa::input]`.
/// These fields can only be mutated by an explicit call to a setter
/// with an `&mut` reference to the database,
/// and therefore cannot be mutated during a tracked function or in parallel.
/// This makes the implementation considerably simpler.
pub struct InputFieldIngredient<K, F> {
```